pkg/config/middleware_test.go

```diff
@@ -28,3 +34,95 @@ func TestSetupGlobalMiddleware(t *testing.T) {
 	assert.NotNil(t, hh)
 	Config.PProfEnabled = true
 }
+
+func TestAuthMiddleware(t *testing.T) {
+	var h, hh http.Handler
+
+	t.Run("it will redirect if jwt enabled but no cookie passed", func(t *testing.T) {
+		Config.JWTAuthEnabled = true
+		hh = SetupGlobalMiddleware(h)
+
+		res := httptest.NewRecorder()
+		res.Body = new(bytes.Buffer)
+		req, _ := http.NewRequest("GET", "http://localhost:18000/api/v1/flags", nil)
+		hh.ServeHTTP(res, req)
+		assert.Equal(t, 307, res.Code)
+
+		Config.JWTAuthEnabled = false
+	})
+
+	t.Run("it will redirect if jwt enabled with wrong cookie passed", func(t *testing.T) {
+		Config.JWTAuthEnabled = true
+		hh = SetupGlobalMiddleware(h)
+
+		res := httptest.NewRecorder()
+		res.Body = new(bytes.Buffer)
+		req, _ := http.NewRequest("GET", "http://localhost:18000/api/v1/flags", nil)
+		req.AddCookie(&http.Cookie{
+			Name:  "access_token",
+			Value: "invalid_jwt",
+		})
+		hh.ServeHTTP(res, req)
+		assert.Equal(t, 307, res.Code)
+
+		Config.JWTAuthEnabled = false
+	})
+
+	t.Run("it will pass if jwt enabled with correct cookie passed", func(t *testing.T) {
+		Config.JWTAuthEnabled = true
+		hh = SetupGlobalMiddleware(h)
+
+		res := httptest.NewRecorder()
+		res.Body = new(bytes.Buffer)
+		req, _ := http.NewRequest("GET", "http://localhost:18000/api/v1/flags", nil)
+		req.AddCookie(&http.Cookie{
+			Name:  "access_token",
+			Value: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJmbGFncl91c2VyIjoiMTIzNDU2Nzg5MCJ9.CLXgNEtwPCqCOtUU-KmqDyO8S2wC_G6PZ0tml8DCuNw", // {"flagr_user": "1234567890"}
+		})
+		hh.ServeHTTP(res, req)
+		assert.NotEqual(t, 307, res.Code)
```

Comment:
Nit, might be better to `assert.Equal` the actual value we get, so that tests document behavior more directly.